### ISSUE TITLE
Possible false positives in UseCodeInterface check for bulk_actions.

### ISSUE THREAD

**Issue by jeroen11dijk** [leonqadirie/ash_credo#111]
Hello,

First of all I really like this package so thanks for making it. But today I looked into the UseCodeInterface check which looks great but I got a warning for the following segment:

```elixir
      Ash.bulk_create(zone_inputs, GlobalZone, :create,
        return_records?: true,
        return_errors?: true,
        authorize?: false
      )
```

```elixir
┃ [F] → Prefer `Zelo.Geography.GlobalZone.create` over `Ash.bulk_create/4`.
┃       lib/zelo/geography/road_works_importer.ex:478:7 #(Zelo.Geography.RoadWorksImporter.bulk_insert)
```

Is this a false positive or is there a way to do this action in bulk in a different way that is preferred?

**Comment by leonqadirie** [comment]
Thanks for the report, at the very least the suggestion is ~wrong~ not great, I’ll look into it.

Is this used in a test?

In any case please bear with me, likely can’t dive into it today as we have a big work deadline.

**Comment by jeroen11dijk** [comment]
No this is my actual code, for checks like these I generally ignore my test files. Also Im not in a hurry but thought that it would be a good thing to discuss!

**Comment by leonqadirie** [comment]
Always good to discuss to improve the tools or learn about them!

I will definitely improve the suggestion, but IMO the check is correct. Code interfaces support lists/streams and in the update/destroy case also queries, and if you need bulk action-specific options, there is `bulk_options`; cf. https://hexdocs.pm/ash/3.2.2/code-interfaces.html#bulk-atomic-actions. 

So assuming you still want to keep the `authorize?: false`, you could have a code interface like so:

```elixir
Zelo.Geography.GlobalZone.create(zone_inputs,
    authorize?: false,
    bulk_options: [
      return_records?: true,
      return_errors?: true
    ]
  )
```

Anything I'm missing?

**Comment by jeroen11dijk** [comment]
I agree with that and that does work! As said I was wondering if there was just a smarter way to do it and a better suggestion would work with this

**Comment by leonqadirie** [comment]
Thank you for the nudge, I hope `v0.11` explains this better :)
